### PR TITLE
[macOS] Remove Pre-Monterey-specific configuration

### PR DESCRIPTION
**Pull request by litherum**
#### bdebf45118173d2bf7784da4ac4708cd29f4bfe3
<pre>
[macOS] Remove Pre-Monterey-specific configuration
<a href="https://bugs.webkit.org/show_bug.cgi?id=258400">https://bugs.webkit.org/show_bug.cgi?id=258400</a>
rdar://111150889

Reviewed by Alexey Proskuryakov.

All supported macOSes are Monterey or later.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:

Canonical link: <a href="https://commits.webkit.org/265432@main">https://commits.webkit.org/265432@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/93c8fd7fa958334595828cae0431edab317696bb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10874 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11090 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11398 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12522 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10410 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13464 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11069 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13321 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11033 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11938 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9161 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12926 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9231 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9819 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17061 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/9210 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10302 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9971 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13215 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10316 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/10428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8512 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/10984 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9593 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2998 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2610 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13866 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11292 "Built successfully") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/10289 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2762 "Passed tests") |

### CODE REVIEW

File: Source/WTF/wtf/PlatformEnableCocoa.h [hunk 1]
```diff
@@ -794,7 +794,7 @@
 #endif
 
 #if (!defined(ENABLE_APP_HIGHLIGHTS) \
-    && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) \
+    && (PLATFORM(MAC) \
         || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 150000) \
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformEnableCocoa.h]: Don't need this check either (there are more iOS version checks below, which I won't comment on).

File: Source/WTF/wtf/PlatformEnableCocoa.h [hunk 2]
```diff
@@ -769,23 +769,23 @@
...
+#if !defined(ENABLE_SET_WEBCONTENT_PROCESS_INFORMATION_IN_NETWORK_PROCESS) && PLATFORM(MAC)
 #define ENABLE_SET_WEBCONTENT_PROCESS_INFORMATION_IN_NETWORK_PROCESS 1
 #endif
 
 #if !defined(ENABLE_WEBM_FORMAT_READER) && PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300
 #define ENABLE_WEBM_FORMAT_READER 1
 #endif
 
-#if !defined(ENABLE_OPUS) && (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000 || PLATFORM(IOS) || PLATFORM(IOS_SIMULATOR) || PLATFORM(VISION))
+#if !defined(ENABLE_OPUS) && (PLATFORM(MAC) || PLATFORM(IOS) || PLATFORM(IOS_SIMULATOR) || PLATFORM(VISION))
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformEnableCocoa.h]: Suspicious about lack of tvOS and macCatalyst. Also, `IOS_SIMULATOR` is superfluous IIRC.

File: Source/WTF/wtf/PlatformHave.h [hunk 3]
```diff
@@ -532,11 +534,11 @@
 #define HAVE_DEVICE_MANAGEMENT 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) || PLATFORM(IOS_FAMILY)
+#if PLATFORM(MAC) || PLATFORM(IOS_FAMILY)
 #define HAVE_NSURLSESSION_WEBSOCKET 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) || PLATFORM(IOS_FAMILY)
+#if PLATFORM(MAC) || PLATFORM(IOS_FAMILY)
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformHave.h]: Is this just `PLATFORM(COCOA)` now?

File: Source/WTF/wtf/PlatformHave.h [hunk 4]
```diff
@@ -886,7 +888,7 @@
 #define HAVE_AVCONTENTKEYREQUEST_COMPATABILITIY_MODE 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) \
+#if PLATFORM(MAC) \
     || PLATFORM(IOS_FAMILY)
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformHave.h]: Also COCOA, but also this is now formatted super badly.

File: Source/WTF/wtf/PlatformHave.h [hunk 5]
```diff
@@ -1266,7 +1269,7 @@
 #define HAVE_UI_CONTEXT_MENU_PREVIEW_ITEM_IDENTIFIER 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 120000) || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED < 150000)
+#if PLATFORM(MAC) || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED < 150000)
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformHave.h]: This change looks incorrect.

**Review by aproskuryakov — APPROVED** [review]
r=me with multiple optional comments. There is one change that looks incorrect, so please address that before landing.

File: Source/WTF/wtf/PlatformEnableCocoa.h [hunk 6]
```diff
@@ -769,23 +769,23 @@
 #define ENABLE_VORBIS 1
 #endif
 
-#if !defined(ENABLE_SET_WEBCONTENT_PROCESS_INFORMATION_IN_NETWORK_PROCESS) && PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000
+#if !defined(ENABLE_SET_WEBCONTENT_PROCESS_INFORMATION_IN_NETWORK_PROCESS) && PLATFORM(MAC)
 #define ENABLE_SET_WEBCONTENT_PROCESS_INFORMATION_IN_NETWORK_PROCESS 1
 #endif
 
 #if !defined(ENABLE_WEBM_FORMAT_READER) && PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformEnableCocoa.h]: This SDK version check is also unneeded now.

File: Source/WTF/wtf/PlatformHave.h (hunk 5 above)
Comment (litherum) [reply Source/WTF/wtf/PlatformHave.h]: Good catch!! Thank you!

File: Source/WTF/wtf/PlatformEnableCocoa.h (hunk 6 above)
Comment (litherum) [reply Source/WTF/wtf/PlatformEnableCocoa.h]: Yep, that's done in https://github.com/WebKit/WebKit/pull/15189

File: Source/WTF/wtf/PlatformEnableCocoa.h (hunk 2 above)
Comment (litherum) [reply Source/WTF/wtf/PlatformEnableCocoa.h]: Right, I think you're right that the original author probably made a mistake here, but I'd like these cleanup patches not to have behavior changes. I think after I'm done with this cleanup project I should audit all the Platform*.h files and try to find all situations like this and see if they can be adjusted.

File: Source/WTF/wtf/PlatformEnableCocoa.h (hunk 1 above)
Comment (litherum) [reply Source/WTF/wtf/PlatformEnableCocoa.h]: Yep, I'll do these in a follow-up.

File: Source/WTF/wtf/PlatformEnableCocoa.h (hunk 2 above)
Comment (aproskuryakov) [reply Source/WTF/wtf/PlatformEnableCocoa.h]: My implication was that a FIXME would be nice, as you added elsewhere.

File: Source/WTF/wtf/PlatformHave.h [hunk 7]
```diff
@@ -532,14 +534,10 @@
 #define HAVE_DEVICE_MANAGEMENT 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) || PLATFORM(IOS_FAMILY)
+#if PLATFORM(MAC) || PLATFORM(IOS_FAMILY)
 #define HAVE_NSURLSESSION_WEBSOCKET 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) || PLATFORM(IOS_FAMILY)
```
Comment (litherum) [inline Source/WTF/wtf/PlatformHave.h]: I think I should put this back and do the accompanying change in a follow-up patch.

File: Source/WTF/wtf/PlatformHave.h [hunk 8]
```diff
@@ -886,13 +884,6 @@
 #define HAVE_AVCONTENTKEYREQUEST_COMPATABILITIY_MODE 1
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) \
-    || PLATFORM(IOS_FAMILY)
-#define HAVE_LEVEL_3_SYSTEM_FONT_WIDTH_VALUES 1
```
Comment (litherum) [inline Source/WTF/wtf/PlatformHave.h]: Ditto.

File: Source/WTF/wtf/PlatformHave.h [hunk 9]
```diff
@@ -1036,16 +1027,12 @@
 #define HAVE_PER_APP_ACCESSIBILITY_PREFERENCES 1
 #endif
 
-#if !PLATFORM(MAC) || __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000
-#define HAVE_APP_SPECIFIC_ACCESSIBILITY_SETTINGS 1
```
Comment (litherum) [inline Source/WTF/wtf/PlatformHave.h]: Ditto.

**Pull request merged**